amazon/ion/reader.py

```diff
@@ -122,12 +88,14 @@ def __init__(self, is_unicode=False):
         self.__size = 0
         self.__data_cls = CodePointArray if is_unicode else bytearray
         if is_unicode:
-            self.__chr = safe_unichr
-            self.__element_type = six.text_type
+            self.__chr = chr
+            self.__element_type = str
         else:
-            self.__chr = chr if six.PY2 else lambda x: x
-            self.__element_type = six.binary_type
-        self.__ord = ord if (six.PY3 and is_unicode) else lambda x: x
+            # TODO: this may be able to be fully eliminated
+            self.__chr = lambda x: x
+            self.__element_type = bytes
+        # TODO: this may be able to be fully eliminated
+        self.__ord = ord if is_unicode else lambda x: x
```

Comment:
Not sure we can eliminate these. My recollection is that we use the BufferQueue for both binary and text data. When it's binary, we need to pass raw bytes back and forth without interpretation. Did you have something else in mind?

Reply:
Honestly I don't know. I added those comments only to hopefully get some input from others and/or as a note for myself to follow up later.

Reply:
Yes. I see that this is used for both binary and text. (I overthought it at first).